src/base/bittorrent/customstorage.h

```diff
@@ -30,10 +30,73 @@
 
 #include <libtorrent/aux_/vector.hpp>
 #include <libtorrent/fwd.hpp>
-#include <libtorrent/storage.hpp>
+#include <libtorrent/version.hpp>
 
 #include <QString>
 
+#if (LIBTORRENT_VERSION_NUM >= 20000)
+#include <libtorrent/disk_interface.hpp>
+#include <libtorrent/file_storage.hpp>
+#include <libtorrent/io_context.hpp>
+
+#include <QHash>
```

Comment:
I think we should still move header inclusions to above.

Reply:
Please clarify. They aren't used in libtorrent-1.2 related implementation.

Reply:
I meant following the style here: https://github.com/qbittorrent/qBittorrent/blob/d2c21ce507bc4188570cca70d740c78b33ce9442/CODING_GUIDELINES.md#7-header-inclusion-order
`// conditional that contains headers from several different header groups`

Reply:
>Is it what you mean now?

It is close to what I have in mind:
```c++
#include <libtorrent/aux_/vector.hpp>
#include <libtorrent/fwd.hpp>
#include <libtorrent/version.hpp>

#if (LIBTORRENT_VERSION_NUM >= 20000)
#include <libtorrent/disk_interface.hpp>
#include <libtorrent/file_storage.hpp>
#include <libtorrent/io_context.hpp>
#else
#include <libtorrent/storage.hpp>
#endif

#include <QString>

#if (LIBTORRENT_VERSION_NUM >= 20000)
#include <QHash>
#endif
```
You can pick one that is suitable.